src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/WasmBrowserTestRunner.cs

```diff
@@ -0,0 +1,190 @@
+                {
+                    webServerCts.Cancel();
+                }
+                driver.Quit();
+            }
+        }
+
+        private async Task RunLogMessagesPump(IWebDriver driver, TaskCompletionSource<ExitCode> taskCompletion, CancellationToken token)
+        {
+            try
+            {
+                while (!token.IsCancellationRequested && !taskCompletion.Task.IsCompleted)
+                {
+                    foreach (var logEntry in driver.Manage().Logs.GetLog(LogType.Browser))
+                    {
+                        if (logEntry.Level == SLogLevel.Severe)
+                        {
+                            // throw on driver errors, or other errors that show up
+                            // in the console log
+                            throw new ArgumentException(logEntry.Message);
+                        }
+
+                        var match = s_consoleLogRegex.Value.Match(Regex.Unescape(logEntry.Message));
+                        string msg = match.Success ? match.Groups[1].Value : logEntry.Message;
+                        msg += Environment.NewLine;
+
+                        _processLogMessage(msg);
+
+                        if (!msg.StartsWith("WASM EXIT "))
+                        {
+                            continue;
+                        }
+
+                        if (int.TryParse(msg.Substring("WASM EXIT ".Length).Trim(), out var code))
+                        {
+                            taskCompletion.SetResult((ExitCode) Enum.ToObject(typeof(ExitCode), code));
+                        }
+                        else
+                        {
+                            _logger.LogDebug($"Got an unknown exit code in msg: '{msg}'");
+                            taskCompletion.SetResult(ExitCode.TESTS_FAILED);
+                        }
+                    }
+
+                    if (driver.CurrentWindowHandle == null) {
+                        // Accessing this property will throw, if chrome has crashed
+                    }
+
+                    await Task.Delay(100, token).ConfigureAwait(false);
+                }
+            }
+            catch (TaskCanceledException)
+            {
+                // Ignore
+            }
+        }
+
+        private string BuildUrl(string webServerAddr)
+        {
+            var uriBuilder = new UriBuilder($"{webServerAddr}/index.html");
+            var sb = new StringBuilder();
+            foreach (var arg in _passThroughArguments)
+            {
+                if (sb.Length > 0)
+                    sb.Append("&");
+
+                sb.Append($"arg={HttpUtility.UrlEncode(arg)}");
+            }
+
+            uriBuilder.Query = sb.ToString();
+            return uriBuilder.ToString();
+        }
+
+        private static async Task<string> StartWebServer(CancellationToken token)
+        {
+            var host = new WebHostBuilder()
+                .UseSetting("UseIISIntegration", false.ToString())
```

Comment:
curious why we need that, shouldn't it be the default?

Reply:
Hm I took the code from what we had for ProxyDriver. This part seems to be from old code. AFAICS, it is the default, so it can be removed. 👍